src/main/scala/firrtl/annotations/Annotation.scala

```diff
@@ -38,6 +38,18 @@ trait Annotation extends Product {
     * @return
     */
   def getTargets: Seq[Target] = extractComponents(productIterator.toIterable).toSeq
+
+  /** Returns a deduplicable representation of this [[Annotation]]: a 3-tuple of the
+    * deduplicated annotation's "dedup key", the deduplicated [[Annotation]], and the
+    * [[firrtl.annotations.ReferenceTarget ReferenceTarget]](s) to the annotated objects.
+    *
+    * If two absolute instances of this [[Annotation]] would deduplicate to the same
+    * local form, both of their "dedup key"s must be equivalent.
+    *
+    * Returning None signifies this annotation will not deduplicate.
+    * @return
+    */
+  private[firrtl] def dedup: Option[((ReferenceTarget, Any), Annotation, ReferenceTarget)] = None
```

Comment:
```suggestion
  private[firrtl] def dedup: Option[(Any, Annotation, ReferenceTarget)] = None
```

Since `(ReferenceTarget, Any)` is itself an `Any`, I think the key should just be `Any`.

Reply:
Should this be private to firrtl? I may be wrong but it seems to me to be pretty important to annotation writers.

It would be useful to have a generic mixin that automagically defines this, by grabbing all the parameters of the case class and filtering out the target.

Reply:
I totally agree this could be important, I don't want it to be public *yet*. I mainly want to get this fix in and released as 1.4.4 while we discuss what the user-facing API should be.